framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,17 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
+        }
         if ($asHtml) {
-            return static::truncateHtml($string, $length, $suffix, $encoding ?: Yii::$app->charset);
+            return static::truncateHtml($string, $length, $suffix, $encoding);
         }
 
-        if (mb_strlen($string, $encoding ?: Yii::$app->charset) > $length) {
-            return rtrim(mb_substr($string, 0, $length, $encoding ?: Yii::$app->charset)) . $suffix;
+        if (mb_strlen($string, $encoding) > $length) {
+            return rtrim(mb_substr($string, 0, $length, $encoding)) . $suffix;
+        } else {
+            return $string;
```

Comment:
What do the above two lines do?

Reply:
If string length is more than limit defined in `$length` it trims it to the length specified, then adding `$suffix`.

Reply:
It's the two new lines

```
} else {
    return $string;
```

that i think need explanation. The next line after the `else {}` is `return $string;`